server/konnectors/maif.js

```diff
@@ -0,0 +1,313 @@
+        callback(err);
+      } else {
+        moment.locale('fr');
+        const importDate = moment().format('LLL');
+        const payload = { profile: JSON.parse(body), date: importDate };
+
+        maifuser.updateAttributes(payload, (err) => { // mise à jour du maifuser en base en insérant le token
+          sendNotification('data retrieved', 'mes-infos-maif');
+          callback(err);
+        });
+      }
+    }, false);
+  });
+}
+
+/**
+* refreshToken function
+* called at each scheduled import (every hour/day/week/month)
+* get new token and refresh token
+* call getToken with token and refresh token
+*/
+function refreshToken(requiredFields, entries, data, next) {
+  MaifUser.first((err, maifUser) => {
+    let tokenValid = true;
+    if (maifUser !== undefined) {
+      const token = maifUser.password;
+      if (token !== undefined) {
+        const options = {
+          url: `${connectUrl}/token`,
+          jar: true,
+          method: 'POST',
+          headers: {
+            Authorization: `Basic ${b64Client}`,
+          },
+          form: {
+            grant_type: 'refresh_token',
+            refresh_token: token,
+          }
+        };
+        request(options, (err, response, body) => {
+          try {
+            JSON.parse(body);
+          } catch (e) {
+            err = 'error';
+          }
+          if (err != null) { // refresh token not valid anymore
+            sendNotification('refresh token not valid', 'konnectors/konnector/maif');
+            next(err);
+          } else {
+            const jsonToken = JSON.parse(body);
+            getToken(jsonToken.token, jsonToken.token_refresh, next);
+          }
+        }, false);
+      } else {
+        tokenValid = false;
+      }
+    } else {
+      tokenValid = false;
+    }
+    if (!tokenValid) {
+      // Maybe we have no token yet !
+      getCode(requiredFields, (err) => {
+        if (err) {
+          sendNotification('refresh token not valid', 'konnectors/konnector/maif');
+          next(err);
+        } else {
+          next();
+        }
+      });
+    }
+  });
+}
+
+/**
+* Display a notification in Cozy
+* code : code of the message to send
+* appToOpen : Link to the app that will be opened on notification's click
+*/
+function sendNotification(code, appToOpen) {
+  code = code === undefined ? '' : code;
+  appToOpen = appToOpen === undefined ? 'konnectors/konnector/maif' : appToOpen;
+  const notifContent = localization.t(code, {});
+  notifHelper.createTemporary({
+    app: 'konnectors',
+    text: notifContent,
+    resource: {
+      app: appToOpen,
+    }
+  });
+}
+
+/**
+* generate UUID for nonce and state parameters
+*/
+function generateUUID() {
+  let d = new Date().getTime();
+  const uuid = 'xxxxxxxx-xxxx-4xxx-yxxx-xxxxxxxxxxxx'.replace(/[xy]/g, (c) => {
+    const r = (d + Math.random() * 16) % 16 | 0;
+    d = Math.floor(d / 16);
+    return (c === 'x' ? r : (r & 0x3 | 0x8)).toString(16);
```

Comment:
Thank's, I didn't notice it was a s-o contribution ^^.
It may mean that we don't need a uuid construct in this way, but just some uuid. So I will make some test with some cleaner and more integrated solutions.
I make a proposition soun.